CriticalMapsKit/Sources/AppFeature/AppFeatureCore.swift

```diff
@@ -320,7 +372,17 @@ public struct AppFeature: ReducerProtocol {
       case let .requestTimer(timerAction):
         switch timerAction {
         case .timerTicked:
-          return EffectTask(value: .fetchData)
+          return .run { send in
+            await withThrowingTaskGroup(of: Void.self) { group in
+              group.addTask {
+                await send(.fetchLocations)
+              }
+              group.addTask {
+                await send(.fetchChatMessages)
```

Comment:
just a question to understand how iOS is doing this: the desired outcome is that chat (post and get) and location (get) will only be run on demand, like only when the user opens that part of the app (map or chat window). the only thing that gets called on an interval (30sec) would be the locations put, is that right for that implementation as well?

Reply:
ok. the chat message get would be run on a timer tick (12 sec) as well as the getting the locations. The map is kind of open all the time due to the navigation structure we have but I could pause the timer when another view is openend of course. 
regarding the chat feature: I can update the behaviour to only fetch messages when the view is open but then we would loose the "unread messages" feature I guess since it wouldn't make a lot sense anymore. But I am not sure if anyone was into that one anyhow 😅

The location put is currently triggered on location updates. If a user moves more then 40 metres the client will send that request.

Reply:
pausing once the map view is out of sight sounds like a good idea. there can be another call right after switching back to the map though.

Reply:
why would the "unread messages" feature not work anymore? wouldn't that just be a matter of comparing the last shown messages to the ones that we added since then? alternatively there could be a timestamp that saves the last shown message. when there is new content everything with an older timestamp would be considered "new". does that make sense or am i missing the point?

Reply:
since we are not getting message updates anymore when the view is not showing there will only be data available the user has already seen. the comparison behaviour from before is very close to what you suggested btw